Match unmatched backticks in compiler/

est31 opened the pull request:
Found with GNU grep:
```
grep -rEn '^(([^`]*`){2})*[^`]*`[^`]*$' compiler/ | rg -v '\s*[//]?.{1,2}```'
```

est31 commented:
These two I don't know what to put:

https://github.com/rust-lang/rust/blob/13471d3b2046cce78181dde6cfc146c09f55e29e/compiler/rustc_hir_typeck/src/closure.rs#L401

https://github.com/rust-lang/rust/blob/13471d3b2046cce78181dde6cfc146c09f55e29e/compiler/rustc_trait_selection/src/solve/assembly.rs#L250

saethlin commented on compiler/rustc_const_eval/src/interpret/intrinsics.rs:
```diff
@@ -457,7 +457,7 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
                 assert!(
                     index < dest_len,
-                    "Index `{}` must be in bounds of vector with length {}`",
+                    "Index `{}` must be in bounds of vector with length `{}`",
```
If these are integers, it would be a bit odd to surround them with backticks, no?

est31 replied:
They are integers, and I agree that it is odd. It's done the same way a bit further down in the file. I didn't want to make changes that are too unrelated. I don't know, I can remove them as well.

saethlin commented on compiler/rustc_mir_transform/src/lower_slice_len.rs:
```diff
@@ -69,7 +69,7 @@ fn lower_slice_len_call<'tcx>(
                     // from something like `_5 = core::slice::<impl [u8]>::len(move _6) -> bb1`
                     // into `_5 = Len(*_6)
-                    // goto bb1
+                    // goto bb1`
```
I don't know what the correct change is here, but I'm pretty sure this isn't it. This implies that the newline after the assignment isn't meaningful, and it is. I would use triple backticks or just none at all, such as

```
                    // from something like
                    // _5 = core::slice::<impl [u8]>::len(move _6) -> bb1
                    // into 
                    // _5 = Len(*_6)
                    // goto bb1
```

est31 committed:
Don't put integers into backticks during formatting

saethlin commented on library/std/src/sys_common/wtf8.rs:
```diff
@@ -594,7 +594,7 @@ impl Wtf8 {
 
     /// Returns the code point at `position` if it is in the ASCII range,
-    /// or `b'\xFF' otherwise.
+    /// or `b'\xFF` otherwise.
```
```suggestion
    /// or `b'\xFF'` otherwise.
```

saethlin commented on library/std/src/sys_common/net/tests.rs:
```diff
@@ -6,7 +6,7 @@ fn no_lookup_host_duplicates() {
     let lh = match LookupHost::try_from(("localhost", 0)) {
         Ok(lh) => lh,
-        Err(e) => panic!("couldn't resolve `localhost': {e}"),
+        Err(e) => panic!("couldn't resolve localhost: {e}"),
```
I think the original intent here is to have some backticks, because we do mean literally the string "localhost" not the abstract idea.

saethlin commented:
I think you should split out the library diff into a separate PR, based on this topic: https://rust-lang.zulipchat.com/#narrow/stream/131828-t-compiler/topic/Spelling.20is.20style.3F/near/328759605

est31 commented:
@saethlin I think the concerns were mostly about PRs, many of which very low effort, that were extremely huge and offloaded the reviewing of the tool to the PR reviewer. They had questionable quality, and sometimes it was questionable if they were improvements at all. Here the improvement is tiny, but very clear. It should be easy to review it. I just saw an error message with an unmatched ` and then wondered about other occurences of the character in the code.

est31 committed:
Rustdoc-ify LiteralKind note

est31 commented:
Okay reading through the thread of #58619, changes that only change rustc-internal comments aren't really liked, but if they are small in size, or easy to review they might still be accepted. Changes to public facing features/docs are still definitely wanted, even if cosmetic.

I've split up this PR into multiple commits:

* "Match end user facing unmatched backticks in compiler/" should be the least controversial. I see ICE messages, diagnostic messages, and lint documentation that gets [rendered to the rustc mdbook](https://doc.rust-lang.org/rustc/lints/listing/index.html) (from my understanding, correct me if I'm wrong) as end user facing.
* "Match unmatched backticks in compiler/ that are part of rustdoc" concerns changes to the rustdoc that is rendered to the publicly available [rustc-internal rustdoc](https://doc.rust-lang.org/nightly/nightly-rustc). Those are rustc developer facing changes.
* "Match unmatched backticks in comments in compiler/" touches non-rustdoc comments only, that are only read by people who have directly opened the file (instead of browsing the rustdoc). This is probably the most controversial of the bunch.
* "Rustdoc-ify LiteralKind note" turns a `//` comment into a rustdoc comment, as well as adjusts an unmatched backtick to use ""s as is done in the surrounding rustdocs.
* "Don't put integers into backticks during formatting" addresses @saethlin 's comments [from above](https://github.com/rust-lang/rust/pull/108685#discussion_r1123929548).

The commits have differing levels of controversialness. Please tell me which ones are not wanted and I should remove, and which ones I should put into different PRs. Thanks!

Nilstrieb commented:
You can make a separate PR with the compiler comments and add me or @WaffleLapkin as the reviewer, we're fine with reviewing it.

est31 commented:
Alright, I have removed even the rustdoc changes from this PR (outside of the one commit about LiteralKind) and moved the two into #108694. Sorry for all the noise, I hope now it should be easy enough to review the patches.

est31 committed:
Match end user facing unmatched backticks in compiler/

petrochenkov commented:
Changes in this PR are certainly not in the same category as in PRs that motivated https://github.com/rust-lang/rust/issues/58619.
Missing left or right quotes a obviously typos/mistakes.
@bors r+

bors commented:
:pushpin: Commit c54f061228b239cb80df2dafd622c5b55513d4f2 has been approved by `petrochenkov`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=petrochenkov c54f061228b239cb80df2dafd622c5b55513d4f2 -->
<!-- homu: {"type":"Approved","sha":"c54f061228b239cb80df2dafd622c5b55513d4f2","approver":"petrochenkov","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

bors commented:
:evergreen_tree: The tree is currently [closed](https://github.com/rust-lang/rust/pull/108677#issuecomment-1453022721) for pull requests below priority 100. This pull request will be tested once the tree is reopened.
